### PR TITLE
Drop deprecated message APIs

### PR DESCRIPTION
**Pull request by kgaillot**
... and make the message code more sane

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's the next batch of drops

File: lib/common/tests/messages/pcmk__new_message_as_test.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    pcmk__xml_free(data);
+
+    assert_non_null(message);
+    assert_string_equal(crm_element_value(message, PCMK_XA_REFERENCE), "rt");
+    assert_string_equal(crm_element_value(message, PCMK__XA_CRM_HOST_TO),
+                        "node1");
+    assert_string_equal(crm_element_value(message, PCMK__XA_CRM_SYS_TO), "rs");
+    assert_non_null(message->children);
+    assert_null(message->children->next);
+    assert_string_equal(message->children->name, PCMK__XE_CRM_XML);
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/tests/messages/pcmk__new_message_as_test.c:79]: bah, compilers getting pickier about char * vs unsigned char *

**Comment by kgaillot** [comment]
updated for review so far

File: lib/common/tests/messages/pcmk__new_message_as_test.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrwahl2) [reply lib/common/tests/messages/pcmk__new_message_as_test.c]: yeah I'm used to it by now, my laptop is F39 and it complains

File: lib/common/ipc_pacemakerd.c [hunk 2]
```diff
@@ -208,6 +208,17 @@ dispatch(pcmk_ipc_api_t *api, xmlNode *reply)
         return ack_status == CRM_EX_INDETERMINATE;
     }
 
+    value = crm_element_value(reply, PCMK__XA_T);
+    if (pcmk__parse_server(value) != pcmk_ipc_pacemakerd) {
+        /* @COMPAT pacemakerd <3.0.0 sets PCMK__VALUE_CRMD as the message type,
```
Comment (nrwahl2) [inline lib/common/ipc_pacemakerd.c:213]: ah just bump the remote protocol version and no more worrying about newer remote nodes :)

actually, "remote clients" in the commit message might mean cluster nodes connecting to the remote server, or remote nodes that are clients to the cluster node's pacemakerd, I don't know. I've never fully understood the remote daemon proxying.

File: lib/common/ipc_schedulerd.c [hunk 3]
```diff
@@ -88,6 +88,15 @@ dispatch(pcmk_ipc_api_t *api, xmlNode *reply)
         return false;
     }
 
+    value = crm_element_value(reply, PCMK__XA_T);
+    if (pcmk__parse_server(value) != pcmk_ipc_schedulerd) {
+        crm_info("Unrecognizable message from schedulerd: "
+                  "unexpected message type '%s'",
```
Comment (nrwahl2) [inline lib/common/ipc_schedulerd.c:94]: indentation, wouldn't bother unless there's something else to change

**Pull request merged**